registration/manager.go

```diff
@@ -0,0 +1,129 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/consul-terraform-sync/client"
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/logging"
+	consulapi "github.com/hashicorp/consul/api"
+)
+
+const (
+	// Service defaults
+	defaultServiceName = "Consul-Terraform-Sync"
+	defaultNamespace   = ""
+
+	// Check defaults
+	defaultCheckName                      = "CTS Health Status"
+	defaultCheckNotes                     = "Check created by Consul-Terraform-Sync"
+	defaultDeregisterCriticalServiceAfter = "30m"
+	defaultCheckStatus                    = consulapi.HealthCritical
+
+	// HTTP-specific check defaults
+	defaultEndpoint      = "/v1/status" // TODO: temporary until /v1/health is implemented
+	defaultMethod        = "GET"
+	defaultInterval      = "10s"
+	defaultTimeout       = "2s"
+	defaultTLSSkipVerify = true
+
+	logSystemName = "registration"
+)
+
+var defaultServiceTags = []string{"cts"}
+
+// RegistrationManager handles the registration of a service and the health checks for that service
+// to Consul.
+type RegistrationManager struct {
+	client  client.ConsulClientInterface
+	service *service
+
+	logger logging.Logger
+}
+
+type RegistrationManagerConfig struct {
+	ID               string
+	Port             int
+	TLSEnabled       bool
+	SelfRegistration *config.SelfRegistrationConfig
+}
+
+type service struct {
+	name      string
+	id        string
+	tags      []string
+	port      int
+	namespace string
+
+	checks []*consulapi.AgentServiceCheck
+}
+
+func NewRegistrationManager(conf *RegistrationManagerConfig, client client.ConsulClientInterface) *RegistrationManager {
```

Comment:
Similar here, would you mind adding comments on this struct and the methods below that are exported?